charlib/characterizer/cell.py

```
@@ -139,7 +139,34 @@ def match_direction(pin_name):
         ## 4. Add as much liberty data as we can right now
         self.liberty = liberty.Group('cell', name)
         self.liberty.add_attribute('area', cell_config.get('area', 0.0), 2)
-        for pin in self.all_pins():
+        for pin, var in feedback_paths.items(): # Add storage groups
+            # Get variable names
+            if pin in [pair.inverting_port_name for pair in self.diff_pairs.values()]:
+                continue # skip inverting pins
+            storage_vars = [var,]
+            for pair in self.diff_pairs.values():
+                if pin == pair.noninverting_port_name:
+                    storage_vars.append(feedback_paths[pair.complement(pin)])
```

Ruff (0.15.17)

[warning] Use `list.extend` to create a transformed list

(PERF401)